### PR TITLE
feat: update Facebook sdk to latest v17.0.2

### PR DESCRIPTION
**Pull request by shwetalodhe123**
1. updated Facebook destination version for v2 .
2. I have found 3 deprecated methods while updating which are mentioned below : 
    - setAdvertiserTrackingEnabled
    - setAdvertiserIDCollectionEnabled
    - setAutoLogAppEventsEnabled
   These methods are now replaced with updated version of them.

3. for iOS 14 onwards updated version of following methods are deprecated  as Facebook is using ATT for tracking user consent : 
    - setAdvertiserTrackingEnabled
    - setAdvertiserIDCollectionEnabled
 also this is handled at code level for lower version than 14, for handling this we are using AdSupport property to get values for above mentioned properties.

### CODE REVIEW

File: Examples/SampleAppObjC/SampleAppObjC/AppDelegate.m [hunk 1]
```diff
@@ -22,12 +24,19 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
...
+   // [[FBSDKSettings sharedSettings] setAdvertiserTrackingEnabled:YES];
+   //added alternative code for this line in applicationDidBecomeActive method
     
-    RSConfig *config = [[RSConfig alloc] initWithWriteKey:@"1wvsoF3Kx2SczQNlx1dvcqW9ODW"];
-    [config dataPlaneURL:@"https://rudderstacz.dataplane.rudderstack.com"];
+    //[[FBSDKSettings sharedSettings] setAdvertiserIDCollectionEnabled:YES];
+    //added alternative code for this line in applicationDidBecomeActive method
+    
+    
+    //[[FBSDKSettings sharedSettings] setAutoLogAppEventsEnabled:YES]; //deprecated
```
[the remark this answers is not in the document]
Comment (shwetalodhe123) [reply Examples/SampleAppObjC/SampleAppObjC/AppDelegate.m:34]: Done.

File: Examples/SampleAppSwift/SampleAppSwift/AppDelegate.swift [hunk 2]
```diff
@@ -51,19 +52,35 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
                 switch status {
                 case .authorized:
                     /// `Get Device Consent`: Starting with iOS 14.5, you will need to set `isAdvertiserTrackingEnabled` and log each time you give a device permission to share data with Facebook. Refer Facebook App Event doc here: https://developers.facebook.com/docs/app-events/getting-started-app-events-ios
-                    Settings.shared.isAdvertiserTrackingEnabled = true
+                    
+                    
+                    //Settings.shared.isAdvertiserTrackingEnabled = true
+                   // Setter for 'isAdvertiserTrackingEnabled' is deprecated: The setAdvertiserTrackingEnabled flag is not used for FBSDK v17+ on iOS 17+ as the FBSDK v17+ now relies on ATTrackingManager.trackingAuthorizationStatus.
```
[the remark this answers is not in the document]
Comment (shwetalodhe123) [reply Examples/SampleAppSwift/SampleAppSwift/AppDelegate.swift:58]: Done

File: .github/workflows/deploy-cocoapods-manual.yml [hunk 3]
```diff
@@ -0,0 +1,22 @@
+name: Manually deploy to Cocoapods
+
+on:
+  workflow_dispatch
+
+jobs:
+  deploy-cocoapods-manual:
+    name: Manually deploy to Cocoapods
+    runs-on: macOS-latest
+    if: startsWith(github.ref, 'refs/heads/master')
```
Comment (1abhishekpandey) [inline .github/workflows/deploy-cocoapods-manual.yml:10]:
```suggestion
    if: startsWith(github.ref, 'refs/heads/main')
```

**Comment by shwetalodhe123** [comment]
Reviewed suggestions

**Pull request merged**